### ISSUE TITLE
3000 DNS queries a minute

### ISSUE THREAD

**Issue by pickerin** [tchellomello/python-ring-doorbell#160]
After upgrading to 0.2.8 of ring-doorbell (in order to get the Oauth2 authentication, I'm now seeing about 3000 DNS queries, per minute, to oauth.ring.com.

**Comment by steve-gombos** [comment]
Is this using it through Home Assistant?

**Comment by pickerin** [comment]
@steve-gombos Yes, this is leveraging your library in the core of Home Assistant.  This wasn't happening until the 0.2.8 library was added.

**Comment by jeromelaban** [comment]
I'm seeing the same behavior, multiple DNS queries per second, as well as many connections to the oauth server:

```
IP X.X> X.53: 13576+ AAAA? oauth.ring.com. (32)
IP X.53 > X.X: 13423 3/0/0 CNAME oauth.ring.com.cdn.cloudflare.net., A 104.17.128.1, A 104.17.127.1 (111)
IP X.53 > X.X: 13576 3/0/0 CNAME oauth.ring.com.cdn.cloudflare.net., AAAA 2606:4700::6811:7f01, AAAA 2606:4700::6811:8001 (135)
```

Each resolution is used to make an https request to the resulting address.

**Comment by albertcard** [comment]
@jeromelaban , I had a temporary fix with '**nscd**' on the server running Homeassistant here: https://github.com/tchellomello/python-ring-doorbell/issues/142#issuecomment-569463411

**Comment by pickerin** [comment]
@albertcard Caching will help a lot, but the TTL on oauth.ring.com is set to 2 seconds.  I'm seeing about 50 / second, so it would cache around 99 of 100 requests made over 2 seconds, but it would still need to make 30 requests a minute...we need to figure out why it has to make so many calls.

**Comment by jes1417** [comment]
I am also seeing 700,000+ requests over the last 24 hours

**Comment by steve-gombos** [comment]
So I setup Pi-hole on my HA to see what DNS queries were going on, but I am not seeing anything close to that volume.  What kind of devices are you running on Ring?  Are you overriding the HA scan interval?

**Comment by jeromelaban** [comment]
I've been using this integration for a while in home assistant, and I used to have changed the `scan_interval` on `camera`, to 300s.

I did not notice that in this PR https://github.com/home-assistant/home-assistant/pull/25534, the scan_interval was moved from `camera` to `ring`. Since then, the refresh interval has been 10 seconds. It must not have been a problem previously but now, auth token are getting refreshed [multiple times from here, per device](https://github.com/home-assistant/home-assistant/blob/1ee299b079b3672aad3dcdce8ce1ecf4e5238d07/homeassistant/components/ring/__init__.py#L87).

Changing `scan_interval` on the ring integration helps significantly for the token refresh count, but it may have other impacts on devices' properties refresh.

**Comment by pickerin** [comment]
I'm running two Ring Floodlight Cams and a Ring Doorbell
I haven't touched the scan interval on the default integrations.

**Comment by pickerin** [comment]
> So I setup Pi-hole on my HA to see what DNS queries were going on, but I am not seeing anything close to that volume. What kind of devices are you running on Ring? Are you overriding the HA scan interval?

Hmmm, I'm not actually running Pi-Hole on my HA installation itself, it's a separate box.  Not sure if that makes a difference or not.

**Comment by jes1417** [comment]
Mine is running in a dedicated pi ![](https://i.imgur.com/wJapkEb.png)

**Comment by steve-gombos** [comment]
I'm not sure if all of my network is using the pihole for DNS....  I am going to review the setup tomorrow.  I see traffic to other ring domains, but not specifically oauth.ring.com, which is weird.

**Comment by pickerin** [comment]
> I'm not sure if all of my network is using the pihole for DNS.... I am going to review the setup tomorrow. I see traffic to other ring domains, but not specifically oauth.ring.com, which is weird.

Steve, thanks so much for digging into this.  Just to be clear, all of the DNS queries we're seeing are coming from HA itself.  I see very little traffic from my Ring devices themselves, or my mobile app when I'm using it.  I see an enormous amount coming from my HA host.  So, you'd just want to confirm what DNS your HA system is using, but if your HA system is also your Pi Hole, you may not see it, since it would probably be using upstream DNS servers, not the Pi Hole running on itself.

**Comment by jeromelaban** [comment]
I've been using this integration for a while in home assistant, and I used to have changed the `scan_interval` on `camera`, to 300s.

I did not notice that in this PR https://github.com/home-assistant/home-assistant/pull/25534, the `scan_interval` was moved from `camera` to `ring`. Since then, the refresh interval has been 10 seconds. It must not have been a problem previously but now, auth token are getting refreshed [multiple times from here, per device](https://github.com/home-assistant/home-assistant/blob/1ee299b079b3672aad3dcdce8ce1ecf4e5238d07/homeassistant/components/ring/__init__.py#L87).

Changing `scan_interval` on the ring integration helps significantly for the token refresh count, but it may have other impacts on devices' properties refresh.

I'm seeing a reduction in traffic only in my dev setup, as there's probably some automation in my main installation that is probably poking the ring properties and forces refreshes.
